data_structures/linked_list/remove_nth_node_from_end_of_list.py

```diff
@@ -0,0 +1,123 @@
+"""Learn more about this algorithm: https://www.geeksforgeeks.org/delete-nth-node-from-the-end-of-the-given-linked-list/"""
+
+from __future__ import annotations
+
+
+class Node:
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.next = None
+
+
+class LinkedListExceptionError(Exception):
+    pass
+
+
+class LinkedList:
+    def __init__(self):  # type: ignore
```

Comment:
```suggestion
    def __init__(self) -> None:
```

Reply:
You will save a lot of code if you create `.__iter__()` and `.__len__()` methods like in
https://github.com/TheAlgorithms/Python/blob/17af6444497a64dbe803904e2ef27d0e2a280f8c/data_structures/linked_list/singly_linked_list.py#L37

Reply:
I'll keep this in mind moving forward and take a look at thos.